core/src/test/java/cz/xtf/core/openshift/ClusterVersionOpenShiftBinaryPathResolverTest.java

```diff
@@ -6,34 +6,61 @@
 
 import org.apache.commons.codec.digest.DigestUtils;
 import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.Test;
 
 import cz.xtf.core.config.OpenShiftConfig;
+import cz.xtf.core.config.XTFConfig;
 
 public class ClusterVersionOpenShiftBinaryPathResolverTest {
 
     private OpenShiftBinaryPathResolver resolver = new ClusterVersionOpenShiftBinaryPathResolver();
 
+    @AfterEach
+    public void cleanVersion() {
+        System.clearProperty(OpenShiftConfig.OPENSHIFT_VERSION);
+        XTFConfig.loadConfig();
+        ClusterVersionInfoFactory.INSTANCE.getClusterVersionInfo(true);
+    }
+
     @Test
     public void resolveTest() {
-        String path = resolver.resolve();
+        final String ocpVersion = "4.8.14";
+        setOCPVersion(ocpVersion);
+        testDownloadedVersion("https://mirror.openshift.com/pub/openshift-v4/amd64/clients/ocp/" + ocpVersion
+                + "/openshift-client-linux.tar.gz", ocpVersion).assertAll();
+    }
 
-        SoftAssertions softAssertions = new SoftAssertions();
+    @Test
+    public void resolveNotFoundVersionTest() {
+        final String ocpVersion = "1.1.1";
+        setOCPVersion(ocpVersion);
+        testDownloadedVersion(
+                "https://mirror.openshift.com/pub/openshift-v4/amd64/clients/ocp/stable/openshift-client-linux.tar.gz",
+                ocpVersion).assertAll();
+    }
 
+    private SoftAssertions testDownloadedVersion(String downloadUrl, String ocpVersion) {
+        SoftAssertions softAssertions = new SoftAssertions();
         // path is not null
-        softAssertions.assertThat(path).isNotNull();
+        String resolvedPath = resolver.resolve();
+        softAssertions.assertThat(resolvedPath).isNotNull();
 
         // path is correct and binary file exists
         Path ocPath = Paths.get("tmp/oc/oc");
-        softAssertions.assertThat(path).isEqualTo(ocPath.toAbsolutePath().toString());
+        softAssertions.assertThat(resolvedPath).isEqualTo(ocPath.toAbsolutePath().toString());
         softAssertions.assertThat(Files.exists((ocPath))).isTrue();
 
         // archive is in cache
-        String urlHash = DigestUtils.md5Hex(
-                "https://mirror.openshift.com/pub/openshift-v4/amd64/clients/ocp/4.8.14/openshift-client-linux.tar.gz");
-        Path cachedPath = Paths.get(OpenShiftConfig.binaryCachePath(), "4.8.14", urlHash, "oc.tar.gz");
+        String urlHash = DigestUtils.md5Hex(downloadUrl);
+        Path cachedPath = Paths.get(OpenShiftConfig.binaryCachePath(), ocpVersion, urlHash, "oc.tar.gz");
         softAssertions.assertThat(Files.exists(cachedPath)).isTrue();
+        return softAssertions;
```

Comment:
just call `.assertAll()` here, instead of returning it.

Reply:
done